graph/patterns/alternate-key.md

```diff
@@ -0,0 +1,46 @@
+# Alternate Key Pattern
+
+Microsoft Graph API Design Pattern
+
+_The Alternate Key Pattern provides the ability to query for a single, specific resource identifiable through an alternative attribute (called key) that is not its unique identifier_
+
+## Problem
+
+---
+
+All entities in our system are identified by an [UUID (Universally Unique Identifier)](https://en.wikipedia.org/wiki/Universally_unique_identifier) - which guarantees uniqueness. Often though, that same resource can also be uniquely identified by an alternative, more convenient attribute that provides a better developer experience.
```

Comment:
IMO this statement reads like there is a single UUID for all entities while there is no uniqueness guaranteed outside of an entity type.

Reply:
That is what I meant and I hope it was the case, but I do see how it might be not. I have reworded it in 81ca72a, let me know if that works better!